suite-native/transactions/src/components/TransactionDetail/TransactionDetailInputsSheet.tsx

```diff
@@ -1,110 +1,97 @@
+import { TransactionsRootState } from '@suite-common/wallet-core';
+import { SettingsSliceRootState } from '@suite-native/module-settings';
 
 import { TransactionDetailSheet } from './TransactionDetailSheet';
+import { selectTransactionInputAndOutputTransfers } from '../../selectors';
+import { TransactionDetailInputsSheetSection } from './TransactionDetailInputsSheetSection';
 
 type TransactionDetailInputsSheetProps = {
     isVisible: boolean;
-    transaction: WalletAccountTransaction;
+    txid: string;
+    accountKey: AccountKey;
     onSheetVisibilityChange: () => void;
 };
 
-type TransactionAddressAmountProps = {
-    address: string;
-    amount?: string;
-    symbol: NetworkSymbol;
+type InputsOutputsMainHeader = {
+    inputsCounts: number;
+    outputsCount: number;
 };
 
-const addressAmountColumnStyle = prepareNativeStyle(_ => ({
-    maxWidth: '42.5%',
-}));
+const InputsOutputsMainHeader = ({ inputsCounts, outputsCount }: InputsOutputsMainHeader) => (
+    <Box flexDirection="row" justifyContent="space-between" marginBottom="medium">
+        <Box flex={1} flexDirection="row" alignItems="center" paddingLeft="small">
+            <Text variant="hint" color="textSubdued">
+                Inputs · {inputsCounts}
+            </Text>
+            <Box marginLeft="small">
+                <Icon name="receiveAlt" color="iconSubdued" size="medium" />
+            </Box>
+        </Box>
 
-const TransactionAddressAmount = ({
-    address,
-    amount = '0',
-    symbol,
-}: TransactionAddressAmountProps) => (
-    <Box>
-        <AccountAddressFormatter value={address} variant="hint" />
-        <CryptoAmountFormatter value={amount} network={symbol} isBalance={false} variant="label" />
+        <Box flex={1} flexDirection="row" alignItems="center" paddingLeft="large">
+            <Text variant="hint" color="textSubdued">
+                Outputs · {outputsCount}
+            </Text>
+            <Box marginLeft="small">
+                <Icon name="sendAlt" color="iconSubdued" size="medium" />
+            </Box>
+        </Box>
     </Box>
 );
 
 export const TransactionDetailInputsSheet = ({
     isVisible,
     onSheetVisibilityChange,
-    transaction,
+    txid,
+    accountKey,
 }: TransactionDetailInputsSheetProps) => {
-    const { applyStyle } = useNativeStyles();
+    const transactionTransfers = useSelector(
+        (state: TransactionsRootState & FiatRatesRootState & SettingsSliceRootState) =>
+            selectTransactionInputAndOutputTransfers(state, txid, accountKey),
+    );
 
-    const txInputs = transaction.details.vin;
-    const txOutputs = transaction.details.vout;
+    if (G.isNull(transactionTransfers)) return null;
+    const { externalTransfers, internalTransfers, tokenTransfers } = transactionTransfers;
+    const inputsCount = externalTransfers.reduce((sum, { inputs }) => sum + inputs.length, 0);
+    const outputsCount = externalTransfers.reduce((sum, { outputs }) => sum + outputs.length, 0);
```

Comment:
You could just create a function that you pass inputs / outputs and it returns count so you don't rewrite it twice

Reply:
If there is no other use case for `selectTransactionInputAndOutputTransfers`, you could also return this value from there

Reply:
Or create a new one that uses the previous selector

Reply:
new func created in 70e3a7bc9